src/mem.rs

```diff
@@ -218,6 +234,21 @@ impl Compress {
         self.inner.total_out
     }
 
+    /// Specifies the compression dictionary to use.
+    ///
+    /// Returns the Adler-32 checksum of the dictionary.
+    #[cfg(feature = "zlib")]
+    pub fn set_dictionary(&mut self, dictionary: &[u8]) -> u32 {
+        let stream = &mut *self.inner.stream_wrapper;
+        let rc = unsafe {
+            ffi::deflateSetDictionary(stream, dictionary.as_ptr(), dictionary.len() as ffi::uInt)
+        };
+
+        assert_eq!(rc, ffi::MZ_OK);
```

Comment:
I was thinking it would be better to return `Result<u32, DecompressError>` from this function so that error in case of incorrect dictionary or in case of calling `set_dictionary` at incorrect time (it has to be called only immediately after `Z_NEED_DICT`) could be still handled gracefully.